tests/Moq.Tests/EventHandlersFixture.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD, and Contributors.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System;
+
+using Xunit;
+
+namespace Moq.Tests
+{
+	public class EventHandlersFixture
+	{
+		[Fact]
+		public void Raising_event__directly_on_mock_object__does_not_trigger_handler__if_CallBase_false()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.False(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__triggers_handler__if_CallBase_true()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>() { CallBase = true };
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.True(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__does_not_trigger_handler__if_setup_without_CallBase_present()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.SetupAdd(m => m.Event += It.IsAny<Action>());
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.False(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__triggers_handler__if_setup_with_CallBase_present()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.SetupAdd(m => m.Event += It.IsAny<Action>()).CallBase();
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
```

Comment:
I have to admit it completely escapes me why the `Callbase()` makes a difference here,

Reply:
As explained below: By using `.Callbase()`, you make sure that when you subscribe to the event, the real `+=` accessor gets used (as opposed to mock's parallel mechanism). So when you raise the real event (through the base class' `RaiseEvent` method), the handler should be triggered. This wouldn't be the case if the handler were subscribed to the mock's "fake" event.